src/Rand.cpp

```diff
@@ -19,28 +19,16 @@ int **SamplingQueue = nullptr;
 
 double ranf(void)
 {
-	int32_t k, s1, s2, z;
-	unsigned int curntg;
+	unsigned int tn;
 
 #ifdef _OPENMP
-	curntg = CACHE_LINE_SIZE * omp_get_thread_num();
+	tn = CACHE_LINE_SIZE * omp_get_thread_num();
 #else
-	curntg = 0;
+	tn = 0;
 #endif
-	s1 = Xcg1[curntg];
```

Comment:
Yes, no problem. I've been pretty busy too recently, so not had a chance to put much time in either. Apologies for the bug there, I should have spotted that one. I've found what hopefully is a neater solution in my latest update. Would be nice to test on a machine with a decent number of cores, which would no doubt have caught that issue at the test stage.

I've rebased the code to the latest upstream, and then updated it to fix the bug that you've mentioned. I'm waiting to see the outcome of the automated tests on here at the moment, but it passes the tests here.

Reply:
Thanks @StevenWhitehouse - one more comment on the latest change on just variable style - I think it might be clearer if we use a macro consistently for this, and CAPS it, rather than having omp_get_thread_num() itself potentially macroed - eg: at the top:

#ifdef _OPENMP
  #include <omp.h>
  #define OMP_GET_THREAD_NUM omp_get_thread_num()
#else
  #define OMP_GET_THREAD_NUM 0
#endif

then on 24, 161, 166, 168 (I think) - refer to OMP_GET_THREAD_NUM instead of the function. I think that might be a bit cleaner.

Reply:
I'll test the full 8-core report-9 runs are unaffected before merging...

Reply:
The alternative solution would be to have:

int omp_get_thread_num(void) { return 0; }

rather than a macro. I don't think it matters too much though, and I'm quite happy to make that change as you suggest. I did debate doing that with myself instead of the macro.

Reply:
I think it's cleaner if `omp_get_thread_num()` whenever used refers to the original function in <omp.h>, and we have something different that returns the zero.

Reply:
No problem. That has been testing while I went out for my walk just now, and has just completed. So I've pushed that update just a moment ago.